ksql-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -117,6 +125,9 @@ public void close() {
       log.error("Could not clean up the query with application id: {}. Query status is: {}",
                 queryApplicationId, kafkaStreams.state());
     }
+    if (queryStateListener.isPresent()) {
```

Comment:
replace with `queryStateListener.ifPresent(QueryStateListener::close);`

Reply:
Made the change.